src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -302,6 +301,27 @@ public void ConstructWithInvalidProperties()
             }
         }
 
+        [Fact]
+        public void ConstructWithMultipleEntryPoints()
```

Comment:
I'd also like to see same-project-different-properties as a test case.

Reply:
Oooh, and a test with `GlobalPropertiesToRemove` causing convergence, ie the "agnostic project" scenario.